Dockerfile

```diff
@@ -28,11 +28,47 @@ RUN apt-get update && apt-get install -y --no-install-recommends \
     fonts-noto \
     fonts-noto-color-emoji \
     librsvg2-bin \
+    npm \
+    libnss3 \
+    libnspr4 \
+    libdbus-1-3 \
+    libatk1.0-0 \
+    libatk-bridge2.0-0 \
+    libcups2 \
+    libdrm2 \
+    libxkbcommon0 \
+    libxcomposite1 \
+    libxdamage1 \
+    libxfixes3 \
+    libxrandr2 \
+    libgbm1 \
+    libasound2 \
+    libxshmfence1 \
     && rm -rf /var/lib/apt/lists/* \
     && fc-cache -fv
 
+# Set a shared cache directory for Puppeteer's browser download
+ENV PUPPETEER_CACHE_DIR /usr/local/share/puppeteer_cache
+
+RUN npm install --global mermaid-filter
 RUN cargo install mdbook-pandoc
 
+COPY puppeteer-config.json /etc/puppeteer-config.json
+
+# Define the path to the mmdc executable
+ARG MMD_PATH="/usr/local/lib/node_modules/mermaid-filter/node_modules/.bin"
+
+# Wrap the real mmdc executable to inject the config file option
+RUN mv "${MMD_PATH}/mmdc" "${MMD_PATH}/mmdc-original" && \
+    echo "#!/bin/sh" > "${MMD_PATH}/mmdc" && \
+    echo "exec \"${MMD_PATH}/mmdc-original\" --puppeteerConfigFile /etc/puppeteer-config.json \"\$@\"" >> "${MMD_PATH}/mmdc" && \
+    chmod +x "${MMD_PATH}/mmdc"
+
+RUN useradd --create-home --shell /bin/bash appuser
+RUN chown -R appuser:appuser /book
+RUN chmod 644 /etc/puppeteer-config.json
+USER appuser
```

Comment:
Is this a requirement of running mdbook, or a way to start a new bash environment? It doesn't look like `puppeteer-config.json` specifies any particular user. Oh, maybe it's just so it's run as a non-root user?